workflow/parser.py

```diff
@@ -0,0 +1,67 @@
+import os
+import subprocess
+
+
+class Parser:
+    def __init__(self, params):
+        self.params = params
+
+    def parse(self):
+
+        '''
+        - parse input config file 
+        - build TPS call 
+        - call TPS 
+        - gather outputs 
+        - return:
+            - list of outputs 
+            - output_folder
+            - out_label
+
+      
+        '''
+        # # grab TPS self.params 
+        build = []
+        OUT_FOLDER = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        OUT_LABEL = ""
+
+        # add TPS run call as first arg
+        build.extend(["bash", "./scripts/run"])
+
+        # grab self.params from config
+        req = self.params[1]["TPS"][0]["required"]
+        op = self.params[1]["TPS"][1]["optional"]
+
+        # filter given args
+        filtered = {**req, **op}
+        args = {key:val for key, val in filtered.items() 
+                if filtered[key] != ""}
+
+                    
+        for key, val in args.items():
+            if key == "outlabel":
+                OUT_LABEL = str(val)
+            if key == "outfolder":
+                OUT_FOLDER = os.path.abspath(val)
+            if val == "1":
+                build.extend(['--' + key])
+            else:
+                k = "--" + key
+                build.extend([k, str(val)])
+
+ 
+        print("outfolder switch {}".format(OUT_FOLDER))
+        print("OUTLABEL: ", OUT_LABEL)
+
```

Comment:
Printing the build command here may be helpful for debugging purposes.

Reply:
> I ran into a few problems when I tested locally.
> 
> When I first ran the workflow, there was an error because `import yaml` failed. We'll have to add `pyyaml` to `minimal_env.yml`.
> 
> I tried putting double quotes around by my Cytoscape path at first, and that gave the error:
> 
> ```
>   File "C:\Users\agitter\.conda\envs\py2cyto_environment\lib\site-packages\yaml\scanner.py", line 1224, in scan_flow_scalar_non_spaces
>     "found unknown escape character %r" % ch, self.get_mark())
> yaml.scanner.ScannerError: while scanning a double-quoted scalar
> ```
> 
> The output label I selected seems to be causing problems
> 
> ```
> ls: cannot access 'agitter-test*': No such file or directory
> outfolder switch C:\Users\agitter\Desktop\postdoc\egfr\src\tps
> OUTLABEL:  agitter-test
> Traceback (most recent call last):
>   File "run.py", line 121, in <module>
>     main(sys.argv)
>   File "run.py", line 38, in main
>     outputs, out_folder, label = parser.parse()
>   File "C:\Users\agitter\Desktop\postdoc\egfr\src\tps\workflow\parser.py", line 63, in parse
>     outputs = str(subprocess.check_output(["ls", "-d", str(OUT_LABEL)+"*"], cwd=OUT_FOLDER), encoding= "utf-8").strip().split()
>   File "C:\Users\agitter\.conda\envs\py2cyto_environment\lib\subprocess.py", line 356, in check_output
>     **kwargs).stdout
>   File "C:\Users\agitter\.conda\envs\py2cyto_environment\lib\subprocess.py", line 438, in run
>     output=stdout, stderr=stderr)
> subprocess.CalledProcessError: Command '['ls', '-d', 'agitter-test*']' returned non-zero exit status 2.
> ```
> 
> We can avoid using `ls` for that line and use Pythons `os.path` or `glob` commands instead, right? However, that seems to indicate some other issue, like maybe the TPS command didn't run and generate the expected output files.
> 
> I left a few other comments but didn't review all the files yet.

From what I have tested so far, all file paths in the config file should not have quotes. In my version I don't have quotes on the path to Cystoscope. If I do put quotes, I need to make sure that double back slashes for windows